File: src/com/l2kt/gameserver/network/clientpackets/FinishRotating.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.network.serverpackets.StopRotation;

public final class FinishRotating extends L2GameClientPacket
{
	private int _degree;
	@SuppressWarnings("unused")
	private int _unknown;
	
	@Override
	protected void readImpl()
	{
		_degree = readD();
		_unknown = readD();
	}
	
	@Override
	protected void runImpl()
	{
		final Player activeChar = getClient().getActiveChar();
		if (activeChar == null)
			return;
		
		activeChar.broadcastPacket(new StopRotation(activeChar.getObjectId(), _degree, 0));
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/FinishRotating.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.network.serverpackets.StopRotation

class FinishRotating : L2GameClientPacket() {
    private var _degree: Int = 0
    private var _unknown: Int = 0

    override fun readImpl() {
        _degree = readD()
        _unknown = readD()
    }

    override fun runImpl() {
        val activeChar = client.activeChar ?: return

        activeChar.broadcastPacket(StopRotation(activeChar.objectId, _degree, 0))
    }
}